packages/in_app_purchase/in_app_purchase_storekit/shared/Classes/InAppPurchasePlugin.m

```diff
@@ -340,6 +341,7 @@ - (void)stopObservingPaymentQueue:(FlutterResult)result {
 }
 
 - (void)registerPaymentQueueDelegate:(FlutterResult)result {
+#if TARGET_OS_IOS
```

Comment:
Same here and below; having these silently no-op seems like a confusing and potentially bug-inducing behavior.